### DISCUSSION TITLE
Update guidance around the expanded repeat nomenclature

### DISCUSSION THREAD

**Discussion by Peter-J-Freeman** [HGVSnomenclature/hgvs-nomenclature#2] [HGVS nomenclature proposals]
**We are seeing the expanded repeat nomenclature more and more**
[current guidelines](http://varnomen.hgvs.org/recommendations/DNA/variant/repeated/) have been described by users as being difficult to follow and they are not sure when to apply them.

**Describe the solution you'd like**
A review of the guidelines and clear and concise description of how to apply different rules in different circumstances. It is possible that we may need to expand the rules in certain instances which @jfjlaros may provide.

**Additional context**
We may also wish to consider whether (and if not, under what circumstances) expanded repeat descriptions are annotative to `core` repeat descriptions e.g. del, dup, ins, delins

**An example to get the ball rolling**

I have also noted that the expanded repeat descriptions int their current form do not make it easy to define the "reference" 
state

e.g. from the current website

- NC_000014.8:g.101179660TG[14]

I see that the TG repeat starts at position 101179660 and repeats 14 times, but I have no way of knowing how many repeats there were in the reference sequence. Perhaps the coordinates could be made into a range? Other options?

**Comment by ifokkema** [comment]
Not sure if I'm supposed to dump ideas/suggestions/considerations in here, but another thing that complicates things is the rule that the repeat syntax must contain only sequences divisible by three when coding reference sequences are used. The idea seems logical at first, limiting the repeat syntax to those sequences likely to vary naturally between individuals. However, this means that when mapping variants between the genome and transcripts, situations can occur when a repeat variant on the genome should be a deletion or duplication on the transcript level. If that would then be mapped back to the genome, there's no saying whether that should be a del/dup or rewritten back to a repeat variant syntax.

Is this, by the way, a deliberate limitation to focus on expanding repeats only? Shortened repeats are excluded here?

**Reply by jfjlaros** [reply]
> Not sure if I'm supposed to dump ideas/suggestions/considerations in here

That is pretty much the idea, yes.

> repeat syntax must contain only sequences divisible by three when coding reference sequences are used

I find it difficult to interpret [the note](https://varnomen.hgvs.org/recommendations/DNA/variant/repeated/) on this subject:

> "using a coding DNA reference sequence (“c.” description) a Repeated sequence variant description can be used only for repeat units with a length which is a multiple of 3, i.e. which can not affect the reading frame."

For example, the description `NC_000011.10:g.112088961_112088964CT[8]` seems to be fine, but the equivalent description `NC_000011.10(NM_003002.4):c.264_267CT[8]` may not be. We do not use a coding DNA reference but we do use the `c.` coordinate system.

> Is this, by the way, a deliberate limitation to focus on expanding repeats only? Shortened repeats are excluded here?

Not as far as I am concerned.

**Reply by Peter-J-Freeman** [reply]
> That is pretty much the idea, yes.

Agreed. We are still finding our way r.e. git issues. I am sure we will add some usage guidelines at some point but I guess we need to get a feel for it ourselves. Helps me too because I will likely apply the same guidelines in my work.

> For example, the description NC_000011.10:g.112088961_112088964CT[8] seems to be fine, but the equivalent description NC_000011.10(NM_003002.4):c.264_267CT[8] may not be. We do not use a coding DNA reference but we do use the c. coordinate system.

I think that this is one of the cases where having the reference state clearly provided in the description is essential. I realise this is queried below, but this seems a logical place to respond.

So, the point I want to make is that if we knew the default state of the reference from the description (and in my opinion, a "variant" description should implicitly tell me about the variation that has occurred, it should not require me to open a reference file and start counting ideally). These descriptions provided here are good. They give the range, so the variant is fully described. I can then easily calculate as to whether there are more repeats/less repeats in the default state at the g. level compared to the c. level. This is important since there are still a good number of genes out there, even in GRCh38 which have discrepancy in the alignment lengths between g. and c. and I suspect many of these will be enriched at repetitive sequence, so this could be absolutely relevant as @ifokkema implies

> I am curious as to why this is important though, in many other cases we also do not know the reference state. For deletions, duplications and inversions we only know that something has been deleted, duplicated or inverted, even for deletion-insertions we know that something has been substituted for a given sequence. Only for substitutions, insertions, and repeats using the mixed repeat format, the relevant reference information[1](https://github.com/HGVSnomenclature/HVNC/discussions/2#user-content-fn-1-367d7e0fe7e84623381fe31a78594fa3) is present in the description.

@jfjlaros. Just pulling this point up here.

I see what you are saying, and I should have been more clear r.e. the reference state. What I meant is that I want to know the full range of the variation. In the del, ins, delins etc, format this is implicit from the coordinates, so I know the default state since it is `reference:g.pos1_pos2=`. Identically, the repeat descriptions above which have been provided also provide the information I was referring to, e.g. NC_000011.10:g.112088961_112088964CT[8] and NM_003002.4:c.264_267CT[8], i.e. the reference state is provided by the coordinates. Thanks for the examples. I will refresh my memory r.e. the unique repeat format vs the mixed repeat format.

**Reply by jfjlaros** [reply]
> there are still a good number of genes out there, even in GRCh38 which have discrepancy in the alignment lengths between g. and c.

This is an issue that deserves its own discussion in my opinion.

**Reply by murphyte** [reply]
> there are still a good number of genes out there, even in GRCh38 which have discrepancy in the alignment lengths between g. and c.

Do you mean because of indel diffs between GRCh38 and NMs? Most (but not quite all) current NMs have been updated to match GRCh38, including everything in MANE. Earlier NM versions have more diffs. If you use a different genome (GRCh37, T2T_CHM13v2.0) then they can have differences, and thinking forward to working with more genome for different populations, or a full pan-genome, then differences between a specific genome and a particular (MANE) reference transcript will become more common again. So yes, if there are improvements to be made in HGVS recommendations in this area, then I agree we should have more separate discussion.

**Reply by jfjlaros** [reply]
> Do you mean because of indel diffs between GRCh38 and NMs?

It is highly related, but I was specifically aiming at the practice of using RefSeq transcripts (`NM_...`) and genomic transcripts (`NC_...(NM_...)`) interchangeably. There are variant mapping problems even if the RefSeq transcript aligns perfectly to the genome, see [this issue](https://github.com/mutalyzer/website/issues/19) for an example.

**Reply by Peter-J-Freeman** [reply]
Hi @murphyte . Yes, I do mean the indel differences needed to create alignments for some genes. I totally agree that the work done for GRCh38 has solved a lot of the problems, but not all. Also, GRCh37 is still widely used in clinical genetics. The primary reason is that labs are reluctant to update SOPs until they are forced to, i.e. the clinical accreditation period comes to an end and they need to re-accredit. I really cannot fault the efforts made between GRCh37 and GRCh38 and MANE etc in dealing with this, but there are still issues. We could have done with a GRCh39!!!! Happy to pick this up in a separate discussion. @jfjlaros, do you want me to open a new discussion?

**Reply by Peter-J-Freeman** [reply]
> https://github.com/mutalyzer/website/issues/19

This is an interesting edge case in its own right as well.

**Reply by jfjlaros** [reply]
> do you want me to open a new discussion?

I think that would help to keep things focused.

**Comment by jfjlaros** [comment]
We have been working on the formalisation of the description of (complex) repeated sequences for some time now. The topic caught our interest because of its relation with information theory, in particular data compression. Before going into any discussions, I would first like to share some observations.


**Observations**

Let us assume that there is a reference sequence `R`, with content `GATATG`. For the sake of brevity, I will omit the reference sequence identifier and coordinate system (e.g., instead of `R:g.2_5AT[3]`, I will write `2_5AT[3]`).

***General***

1. There are (at least) [two formats](https://varnomen.hgvs.org/recommendations/DNA/variant/repeated/) for describing repeats: the *unique repeat format* (e.g., `2AT[3]`) and the (perhaps slightly abused) *mixed repeat format* (e.g., `2_5AT[3]`).
2. As far as I can tell, the [prioritisation rule](https://varnomen.hgvs.org/recommendations/general/) does not cover the repeat operator.
3. A repeat contraction can always be described as a deletion.
4. A repeat expansion can be described as a duplication unless the number of added repeat units exceeds the number of repeat units in the reference sequence. E.g., `2_5AT[4]` is equivalent to `2_5dup`, but `2_5AT[5]` has to be described as `5_6insATATAT`.

***Arguably negative aspects of the repeat syntax***

5. It is not clear from a description in *unique repeat format* whether a variant is equivalent to an insertion, a deletion or no change at all.

***Arguably positive aspects of the repeat syntax***

6. It is clear from the notation *what* has been inserted or deleted.
7. It is clear that the reference sequence contains a repeat.
8. It is easier to relate variants to each other. E.g., compare `4_5del`, `2_5dup` and `5_6insATATAT` to `2_5AT[1]`, `2_5AT[4]` and `2_5AT[5]`.
9. The notation is compact when used in conjunction with an insertion (e.g., `1_2insCA[10]`).


**Summary**

- If the repeat operator is given a low priority, it can only be used in edge cases (see points 2, 3 and 4).
- If the repeat operator is given a high priority, the descriptions will convey more information (see points 2, 6, 7 and 8).
- If the *mixed repeat format* is used, it can be determined whether a repeat is contracted or expanded, albeit using some arithmetic (see points 1 and 5).

**Reply by Peter-J-Freeman** [reply]
@jfjlaros. Having looked at the varnomen website (thanks for providing the link) it is not immediately obvious why 2 formats are required. 

I think it would be really useful for this discussion to have clear instructions r.e. the current guidance as to when to apply which type and why. It is not immediately obvious to me why one description type is not sufficient

> It is not clear from a description in unique repeat format whether a variant is equivalent to an insertion, a deletion or no change at all.

Exactly the point I was making above. Thanks

> Arguably positive aspects of the repeat syntax
I agree with the positive points you provided.

> A repeat contraction can always be described as a deletion.

Also, a repeat expansion can also be described as a duplication (or ins if necessary, as you clearly state). This is where my point r.e. annotative descriptions comes in to play. In terms of interoperability (i.e. identification of descriptions in databases and literature), my feeling is that is is better to have a `primary core description` in the format of a dup (expansion of repeats) or a del (contraction of repeats) and a `secondary annotative description` which is the repeat syntax description. It would be interesting to get @ifokkema's  opinion on from the LOVD perspective too. I have seen instances where, since the SPDI paper, users have needlessly tried to apply the repeat syntax where it is not really relevant or required. If the guidelines stated that the repeat syntax is annotative and not primary, this ought not to happen. Would be interested to hear your opinion on this too @jfjlaros

**Reply by jfjlaros** [reply]
> It is not immediately obvious to me why one description type is not sufficient

As far as I know, the *unique repeat format* has been around for a very long time. I presume that the *mixed repeat format* was added to deal with compound repeats, where it may not be clear where one repeat ends and an other begins. To complicate matters even further, combination of these formats have been used in the past. I seem to remember that dbSNP has used a format in which a range was used to indicate the repeat unit (e.g., `2_3AT[3]`, by which `2AT[3]` was meant).

> In terms of interoperability (i.e. identification of descriptions in databases and literature), my feeling is that is is better to have a primary core description in the format of a dup (expansion of repeats) or a del (contraction of repeats) and a secondary annotative description which is the repeat syntax description.

The HGVS nomenclature format may not be the best variant representation format from an interoperability perspective. This is probably why there are multiple machine readable variant formats available. I have always understood that HGVS descriptions are primarily meant to be read by people, so from that perspective, one could argue that all HGVS descriptions are annotative. I also wonder why the line should be drawn here. After all, a duplication is a special case of an insertion, which in itself is a special case of a deletion-insertion.

If people are not only interested in exact matches in a database, but are also interested in certain relationships between variants, I can imagine that people would prefer the repeat syntax (also see **point 8** above).

**Reply by Peter-J-Freeman** [reply]
> As far as I know, the unique repeat format has been around for a very long time. I presume that the mixed repeat format was added to deal with compound repeats, where it may not be clear where one repeat ends and an other begins. To complicate matters even further, combination of these formats have been used in the past. I seem to remember that dbSNP has used a format in which a range was used to indicate the repeat unit (e.g., 2_3AT[3], by which 2AT[3] was meant).

So switching to a single format would not in theory be too complicated? For me it makes sense to do so.

> The HGVS nomenclature format may not be the best variant representation format from an interoperability perspective. This is probably why there are multiple machine readable variant formats available.

Agreed entirely. However, it is the clinical standard and the globally accepted reporting standard in clinical reports and journal entries. So, we have to make HGVS descriptions robust so that they are findable and interoperable  because a huge community use HGVS descriptions as search terms. This is not going away in my opinion unless a new reporting standard comes along. 

> If people are not only interested in exact matches in a database, but are also interested in certain relationships between variants, I can imagine that people would prefer the repeat syntax (also see point 8 above).

I think we need to consider both r.e. my point above this one. I agree with point 8. However, we have prioritisation of variant types already in place. If we include the repeat syntax in the prioritisation, we will need to decide under what circumstances it is used instead of dup or del. What we must consider is that if these rules are put in place, we will likely end up re-naming well documented variants needlessly. So, my preference would be to have the core description as a dup or a del etc (i.e. the simplest possible description), but with additional descriptions to provide the number of repeats if it there is a genuine need to know. However, I also see the argument that it might be simpler to have one description that does everything, but that comes with the aforementioned issues of renaming well characterised variants. That would not necessarily be popular.

On the up side, I think this little section forms the basis of one of the items to put back to the Nomenclature committee

**Reply by jfjlaros** [reply]
> So switching to a single format would not in theory be too complicated?

I do not see any technical objections either.

> So we have to make HGVS descriptions robust so that they are findable and interoperable because a huge community use HGVS descriptions as search terms.

In my experience, interoperability of variant descriptions mainly comes from the surrounding tooling. I suspect that nearly all variant databases that support the HGVS nomenclature rely on HGVS parsers and disambiguation procedures. Furthermore, normalisation will only get us so far when more complex variant descriptions (like allele descriptions) are used.

> If we include the repeat syntax in the prioritisation, we will need to decide under what circumstances it is used instead of dup or del.

I hinted at that in the summary above:

> > - If the repeat operator is given a low priority, it can only be used in edge cases (see points 2, 3 and 4).

So one could argue that this operator should either take precedence over all other operators, or not be used at all. Please note that even if we do choose to make this a high priority operator, there would still be room for discussion. E.g., choices need to be made about the minimal repeat unit length and the minimum number of repeat units present in the reference sequence.

**Reply by ifokkema** [reply]
Regarding having both a del/dup/ins-syntax description as well as a repeat-syntax description for a single variant; I'd just like to note that mixed repeat syntax can easily be converted into del/dup/ins-syntax format without the need for external tooling; i.e., `2_3AT[2]` clearly equals `2_3dup`, but I would only be able to do the reverse conversion using specialized tools (i.e., VV or Mutalyzer). A database platform could convert either the query or the database's contents from mixed repeat syntax to del/dup/ins syntax to match the query to the database contents if needed. However, this would require the use of the mixed repeat syntax over the use of the unique repeat syntax.

**Comment by jfjlaros** [comment]
@Peter-J-Freeman:

> I have also noted that the expanded repeat descriptions int their current form do not make it easy to define the "reference" state

This is only true for the *unique repeat format*, the reference state *can* be deduced when using the *mixed repeat format*.

I am curious as to why this is important though, in many other cases we also do not know the reference state. For deletions, duplications and inversions we only know that *something* has been deleted, duplicated or inverted, even for deletion-insertions we know that *something* has been substituted for a given sequence. Only for substitutions, insertions, and repeats using the *mixed repeat format*, the relevant reference information[^1] is present in the description.

[^1]: This may not be true outside of the context of the HGVS nomenclature, e.g., for fully justified SPDI notations.

**Reply by Peter-J-Freeman** [reply]
@jfjlaros . A note r.e. my concerns is actually on the varnomen web page

`NOTE: a Community Consultation is prepared which will suggest to allow only one format where the entire range of the repeated sequence must be indicated, e.g. g.123_191CAG[23] not g.123CAG[23]`

I gave my reasoning above.

**Reply by ifokkema** [reply]
> @jfjlaros . A note r.e. my concerns is actually on the varnomen web page
> 
> (...)
> 
> I gave my reasoning above.

Considering again that the mixed repeat format allows easy automated conversion into del/dup/ins syntax as well as clarifying to the reader what kind of variant is actually described, I agree with Pete that making the mixed repeat format a requirement for repeat syntax as a whole, would be a good idea.

**Comment by rileygr** [comment]
As a curator working on ClinVar, I agree that improving the guidelines to give clear and concise description of how to apply different rules in different circumstances, would be helpful to the community.  A couple of examples have come to our attention.

**Ranges of repeats**
A description of either an unknown range, where the method does not give an exact repeat number or a known range of repeats that is clinically significant (pathogenic/benign).  It's not entire clear how these should be represented on the website:

An example of the use of square brackets alone for variation in a population is given in the Q&A section of the “Repeated” page, “c.1210-12T[5_9]”, but there are no examples of parentheses with square brackets on that page:
http://varnomen.hgvs.org/recommendations/DNA/variant/repeated/
 
However, there are examples of the use of parentheses with square brackets for repeats on the Alleles page, “NC_000003.12:g.63912687AGC[(50_60)]”, 
https://varnomen.hgvs.org/recommendations/DNA/variant/alleles/

**Different repeat sequence**
There is at least one [documented](https://www.ncbi.nlm.nih.gov/books/NBK564656/#rfc1-canvas.Molecular_Genetics) clinically significant variation where the entire set of AAAAG repeats in the reference is replaced by AAGGG repeats.  An HGVS expression was submitted to ClinVar as a delins: NC_000004.11:g.39350045_39350099delinsAAGGG[141_831].

**Reply by ifokkema** [reply]
> An example of the use of square brackets alone for variation in a population is given in the Q&A section of the “Repeated” page, “c.1210-12T[5_9]”, but there are no examples of parentheses with square brackets on that page: http://varnomen.hgvs.org/recommendations/DNA/variant/repeated/

This page, apparently, is quite outdated. It also uses descriptions like `ins(3)` that on more up-to-date pages are described as `insN[3]`. There are, unfortunately, more inconsistencies on the HGVS varnomen website.


> However, there are examples of the use of parentheses with square brackets for repeats on the Alleles page, “NC_000003.12:g.63912687AGC[(50_60)]”, https://varnomen.hgvs.org/recommendations/DNA/variant/alleles/

As far as I know, `[(50_60)]` is the correct syntax. I understood that this was decided because `50_60` would cause too much confusion with position syntax. LOVD also uses parentheses, and this was discussed extensively with Johan last year. (but not discussed with the rest of the HVNC, I should note)


> **Different repeat sequence** There is at least one [documented](https://www.ncbi.nlm.nih.gov/books/NBK564656/#rfc1-canvas.Molecular_Genetics) clinically significant variation where the entire set of AAAAG repeats in the reference is replaced by AAGGG repeats. An HGVS expression was submitted to ClinVar as a delins: NC_000004.11:g.39350045_39350099delinsAAGGG[141_831].

Assuming that indeed the number of repeats is unknown, the valid HGVS nomenclature for this variant would be `NC_000004.11:g.39350045_39350099delinsAAGGG[(141_831)]`.
